app/validators/answers/number_answer_validator.py

```diff
@@ -71,6 +73,42 @@ def validate_value_in_limits(self):
                 self.MAXIMUM_GREATER_THAN_LIMIT, value=max_value, limit=MAX_NUMBER
             )
 
+        if isinstance(answer_source_max_value, dict):
+            source = answer_source_max_value.get("source")
+            answers_with_context = self.questionnaire_schema.numeric_answer_ranges
+            answer_source_max_value = (
+                self.questionnaire_schema.get_numeric_value_for_value_source(
+                    source, answer_source_max_value, answers_with_context
+                )
+            )
+            if (
+                isinstance(answer_source_max_value, int)
+                and answer_source_max_value > MAX_NUMBER
+            ):
+                self.add_error(
+                    self.MAXIMUM_GREATER_THAN_LIMIT,
+                    value=answer_source_max_value,
+                    limit=MAX_NUMBER,
+                )
+
+        if isinstance(answer_source_min_value, dict):
+            source = answer_source_min_value.get("source")
```

Comment:
Not sure to make this a function since it is repeated?

Reply:
Think it's fine as there's an operator change between these blocks, not just variable changes, so doesn't feel worth the complication to make a function in this scenario. 
Rule of 3 - if you have to repeat it three times, make a function, but don't use as an excuse not to be DRY - Don't Repeat Yourself. In this case it feels alright to keep these repeated parts.

Reply:
Ahh I see, thanks thats a useful mnemonic - I will be using that from now on